### PR TITLE
APP-298-5 Updating and Persisting Customer Details

### PR DESCRIPTION
**Pull request by rcdtroly**
### Why:
`Orders made in the website are still going through even with incomplete details. Or credit card details are not recorded in the system`

### What: 
* For new Customer - create new account in Wordpress before submitting order
* Fix upgrading database
* Remove redundant function

### Refs:
* https://app.asana.com/0/939825968139797/1138124271919163
* https://troly-io.slack.com/archives/C4Y4R7B1T/p1567748657058600

### CODE REVIEW

File: includes/frontend/controllers/class-wp99234-users.php [hunk 1]
```diff
@@ -1101,45 +1115,45 @@ function on_checkout_order_processed( $order_id, $posted ){
...
+        $errors = is_object($data) ? (array)$data->errors : array();
 
-        if( ! empty( $errors ) ){
-            throw new Exception( __( 'An error has occurred, and we could not process your payment. Please ensure your credit card details are correct and try again. You will be contacted via phone ASAP to ensure your order is processed as soon as possible.', 'wp99234' ) );
+        if (!empty($errors)) {
+            throw new Exception(__('An error has occurred, and we could not process your payment. Please ensure your credit card details are correct and try again. You will be contacted via phone ASAP to ensure your order is processed as soon as possible.', 'wp99234'));
 
             ob_start();
-            var_dump( $errors );
+            var_dump($errors);
```
Comment (atifempireone) [inline includes/frontend/controllers/class-wp99234-users.php]: we should not have var_dump in production code

Comment (rcdtroly) [reply includes/frontend/controllers/class-wp99234-users.php]: @atifempireone fixed

**Comment by rcdtroly** [comment]
@atifempireone @JamesEmpireOne i also added CVV in Club registration, and fix not saving CVV during checkout, and logic in `Use existing card` when checked
Alt text

File: includes/frontend/controllers/class-wp99234-users.php [hunk 2]
```diff
@@ -365,11 +365,16 @@ function export_user( $user_id, $load_address = null, $override_data = array(),
...
 
         //Allow CC fields to be updated if checking out and not using the existing card.
-        if( ( defined( 'WOOCOMMERCE_CHECKOUT' ) && WOOCOMMERCE_CHECKOUT ) && ( ! isset( $_POST['use_existing_card'] ) ) ){
-            $meta['cc_name']      = 'cc_name';
-            $meta['cc_number']    = 'cc_number';
-            $meta['cc_exp_month'] = 'cc_exp_month';
-            $meta['cc_exp_year']  = 'cc_exp_year';
+        if (defined('WOOCOMMERCE_CHECKOUT') && WOOCOMMERCE_CHECKOUT) {
+            // Check Credit Card for details
+            if (($_POST['payment_method'] === 'wp99234_payment_gateway' && $_POST['shipping_method'][0] === 'wp99234_shipping_method' && !isset($_POST['wp99234_use_existing_card']))
```
Comment (atifempireone) [inline includes/frontend/controllers/class-wp99234-users.php]: add comment as to why we need this additional check

**Comment by rcdtroly** [comment]
@Ryders @atifempireone  new layout - also using `Card Code` label to cvc to make it consistent (Checkout CC details)
Alt text

**Comment by JamesEmpireOne** [comment]
Test |  Customer does *not* exist in Troly | Customer exists in Troly
------------ | ------------- | ------------- |
One-off order, not signed-in | ? | ?
One-off order, signed-in | ? | ?

## Test breakdown
### One-off order - not signed in

Result | Created | Updated
------------ | ------------- | ------------- |
WP Account  | ? | ?
WC Order | ? | ?
Troly Customer Record | ? | ?
Troly Order Record | ? | ?

### One-off order - not signed in

Result | Created | Updated
------------ | ------------- | ------------- |
WP Account  | ? | ?
WC Order | ? | ?
Troly Customer Record | ? | ?
Troly Order Record | ? | ?

### One-off order - signed in

Result | Created | Updated
------------ | ------------- | ------------- |
WP Account  | ? | ?
WC Order | ? | ?
Troly Customer Record | ? | ?
Troly Order Record | ? | ?

**Pull request merged**